### PR TITLE
Testing with Jest and Selenium's WebdriverJS!

### PR DESCRIPTION
**Pull request by saisankp**
## Purpose
I have added testing, which would be great for our pipeline. I have mainly added two files:
1. ```server.test.js``` which uses Jest and provides some code coverage.
2. ```frontend-test.js``` which uses WebdriverJS.

## Proposed Change

This can be used with our CI/CD pipeline. So far, the tests are basic, such as making sure some frontend Vue.js components are not null etc. This can be improved upon if needed.

## Checklist

- [x] Create some basic tests with Jest!
- [x] Create some basic tests with Selenium-Webdriver!
- [x] Have some sort of code coverage!

## Additional

Please read the ```README.md``` file under ```covid-vue-app/test``` to understand the methodology of testing/approach I took to design this!

## Learning

_Describe the research stage_
Here are some useful links:
https://www.npmjs.com/package/supertest
https://www.npmjs.com/package/selenium-webdriver
https://www.npmjs.com/package/jest

### CODE REVIEW

File: covid-vue-app/test/frontend-test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+const driver = new Builder().forBrowser("firefox").build();
+var numberOfTestsCompleted = 0;
+var numberOfErrors = 0;
+const numberOfTests = 5;
+
+
+async function openTheApp() {
+    try{
+        await driver.get("http://192.168.1.11:8080/");
```
Comment (dabreadman) [inline covid-vue-app/test/frontend-test.js:12]: Change to `localhost` (127.0.0.1), not sure if this would work :L

Comment (saisankp) [reply covid-vue-app/test/frontend-test.js:12]: Sure thing!

**Comment by dabreadman** [comment]
# Please update .gitignore

Unless the `coverage` folders are essential, please update `.gitignore`.

Do `git rm coverage -r -f` to untrack them.

```
# Node ignores
node_modules
build
npm-debug.log

.env
.DS_Store

# Optimised distribution files
dist

# Directory for instrumented libs generated by jscoverage/JSCover
lib-cov

# Coverage files
coverage
```

**Comment by dabreadman** [comment]
Perhaps add some [snapshot testing](https://www.digitalocean.com/community/tutorials/vuejs-jest-snapshot-testing-in-vue).  

More Vue API [here](https://vue-test-utils.vuejs.org/api/#mount)

File: covid-vue-app/test/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## How this app is tested
+
+This application uses Vue.js and Node.js. Testing is done with Jest and Selenium Webdriver. Node.js is tested with Jest, and Vue.js components are tested at ```localhost:8080``` automatically.
```
Comment (dabreadman) [inline covid-vue-app/test/README.md:3]: Why four backticks?

```suggestion
This application uses Vue.js and Node.js. Testing is done with Jest and Selenium Webdriver. Node.js is tested with Jest, and Vue.js components are tested at `localhost:8080` automatically. 
```

File: covid-vue-app/test/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## How this app is tested
+
+This application uses Vue.js and Node.js. Testing is done with Jest and Selenium Webdriver. Node.js is tested with Jest, and Vue.js components are tested at ```localhost:8080``` automatically. 
+
+### server.test.js
```
Comment (dabreadman) [inline covid-vue-app/test/README.md:5]: Maybe format it a little, and list out the tests that was ran in each test?

- Test if `Title` render
- Test if `Cases` is correct

**Review by dabreadman — CHANGES_REQUESTED** [review]
Nice work, could you perhaps change the `README.md` to `TESTING`.MD`?

**Comment by dabreadman** [comment]
And your branch is out-of-date too, merge main :L

**Review by dabreadman — APPROVED** [review]

**Pull request merged**